### DISCUSSION TITLE
can't login with google SSO "auth-error"

### DISCUSSION THREAD

**Discussion by Freundschaft** [outline/outline#1974] [Self Hosting] [answered]
Hello,

I have configured outline to run on my kubernetes cluster and things are looking good, but I can't login, because something fails during oauth login.

I always get redirected to this screen:

![image](https://user-images.githubusercontent.com/5065616/111924887-4961fb00-8aaf-11eb-8bbb-6009a8449d8d.png)

Is there a way how I can get debugging info on why passport fails?

I am running version-0.54.0

Cheers

**Comment by tommoor** [comment]
Errors should be in the server logs output from the Node proccess

**Reply by Freundschaft** [reply]
unfortunately I don't see an obvious error (log attached)
[log.log](https://github.com/outline/outline/files/6178652/log.log)

**Reply by tommoor** [reply]
Indeed, no errors in that snippet – but it's also entirely sql logs. Is todays release, v0.54.0, the first one you've ran?

**Reply by Freundschaft** [reply]
yes, I deleted everything and redeployed from scratch with v0.54.0.
these logs are logs that were generated after trying to log in through google
could the issue somehow be related to redis or S3? or are they unrelated
(you can find the instance at https://outline.lowtechwithrefugees.org/)

**Reply by tommoor** [reply]
There's a slim error it could be S3 if it's failing to upload the avatar, but that _should_ be caught and not result in a failed sign-in.

Have records for teams, users, or user_authentications or authentication_providers been created in the db?

**Reply by Freundschaft** [reply]
> outline=# TABLE teams;
> -[ RECORD 1 ]--+---------------------------------------
> id             | d934b749-7d8c-404a-8695-75d04af8a1aa
> name           | Asso-eko
> slackId        | 
> createdAt      | 2021-03-21 23:36:56.327+00
> updatedAt      | 2021-03-21 23:36:56.358+00
> googleId       | 
> avatarUrl      | https://logo.clearbit.com/asso-eko.org
> deletedAt      | 
> sharing        | t
> subdomain      | asso-eko
> documentEmbeds | t
> guestSignin    | t
> domain         | 
> 

> outline=# TABLE authentication_providers;
> -[ RECORD 1 ]------------------------------------
> id         | ec90f2e3-1d56-45d1-a88b-2ad6d99352af
> name       | google
> providerId | asso-eko.org
> enabled    | t
> teamId     | d934b749-7d8c-404a-8695-75d04af8a1aa
> createdAt  | 2021-03-21 23:36:56.355+00

all other tables you mentioned have been created but are empty

**Reply by tommoor** [reply]
> could the issue somehow be related to redis or S3?

Make sure both are configured, my best guess is S3 right now – it's unfortunate that the error is being swallowed but there is nothing we can do about that until a patch release.

**Reply by Freundschaft** [reply]
Ok will do and report back, thanks for the support so far!

**Reply by Freundschaft** [reply]
> > could the issue somehow be related to redis or S3?
> 
> Make sure both are configured, my best guess is S3 right now – it's unfortunate that the error is being swallowed but there is nothing we can do about that until a patch release.

I tried again today, with minio and redis, and now there is no error message at all, but authentication is still not working. Is there a way, how I can read the error message when authentication fails?

[outline-7dcb6bfb7d-p8c46_logs.txt](https://github.com/outline/outline/files/6185310/outline-7dcb6bfb7d-p8c46_logs.txt)

**Reply by Freundschaft** [reply]
ah wait I cleared the db and tried again and now I see this:

> InvalidAccessKeyId: The Access Key Id you provided does not exist in our records.
> 
>     at Request.extractError (/opt/outline/node_modules/aws-sdk/lib/services/s3.js:700:35)
> 
>     at Request.callListeners (/opt/outline/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
> 
>     at Request.emit (/opt/outline/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
> 
>     at Request.emit (/opt/outline/node_modules/aws-sdk/lib/request.js:688:14)
> 
>     at Request.transition (/opt/outline/node_modules/aws-sdk/lib/request.js:22:10)
> 
>     at AcceptorStateMachine.runTo (/opt/outline/node_modules/aws-sdk/lib/state_machine.js:14:12)
> 
>     at /opt/outline/node_modules/aws-sdk/lib/state_machine.js:26:10
> 
>     at Request.<anonymous> (/opt/outline/node_modules/aws-sdk/lib/request.js:38:9)
> 
>     at Request.<anonymous> (/opt/outline/node_modules/aws-sdk/lib/request.js:690:12)
> 
>     at Request.callListeners (/opt/outline/node_modules/aws-sdk/lib/sequential_executor.js:116:18)
> 
>     at Request.emit (/opt/outline/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
> 
>     at Request.emit (/opt/outline/node_modules/aws-sdk/lib/request.js:688:14)
> 
>     at Request.transition (/opt/outline/node_modules/aws-sdk/lib/request.js:22:10)
> 
>     at AcceptorStateMachine.runTo (/opt/outline/node_modules/aws-sdk/lib/state_machine.js:14:12)
> 
>     at /opt/outline/node_modules/aws-sdk/lib/state_machine.js:26:10
> 
>     at Request.<anonymous> (/opt/outline/node_modules/aws-sdk/lib/request.js:38:9) {
> 
>   code: 'InvalidAccessKeyId',
> 
>   region: null,
> 
>   time: 2021-03-22T23:30:00.298Z,
> 
>   requestId: '166ECDFC6BCE5C3A',
> 
>   extendedRequestId: undefined,
> 
>   cfId: undefined,
> 
>   statusCode: 403,
> 
>   retryable: false,
> 
>   retryDelay: 49.42029345996411
> 
> }

**Reply by Freundschaft** [reply]
OK I fixed this now, Redis and S3 looking good, but still no luck authenticating. (logs attached)

[outline-57c6844bf9-z6k6d_logs.txt](https://github.com/outline/outline/files/6186248/outline-57c6844bf9-z6k6d_logs.txt)

I can't help but think that this might not be related to S3 at all, but some issue with the Google SSO and passport itself.
authentication and user tables are still empty. Shouldn't I see some CREATE or UPDATE commands in the log at some point?

**Reply by andreipoe** [reply]
Hi!

I'm having the same issue you are having: authentication fails, but nothing is printed in the logs. I've gone through the discussion and my database in the same state yours was. As for the `InvalidAccessKeyId` errors, I've noticed these only show up on the first run, and I can use the same credentials in the browser. My `SECRET_KEY` and `UTILS_SECRET` variables are set as suggested (generated using `openssl rand -hex 32`). 

Do you have any more suggestions or hints where I should be looking?

Thanks!

**Reply by tommoor** [reply]
- What version are you using?
- Is it the official docker image?
- Have you setup S3 compatible storage yet?
- Please double check for console errors, you might want to slim down the other logging by clearing the DEBUG env variable incase it's hidden amongst sql logs etc

**Reply by andreipoe** [reply]
Thanks for the quick response!

It's the latest official Docker image, so v0.54. Just like OP, I used the set-up described in the README and minio for S3-like storage. The storage works in the browser and through the CLI tool.

In the docker logs, there is very little output, and no errors at all. I've tried looking through some files in the container, but I couldn't find any more detailed logs. If you can tell me where to look, I'll dig through there and see if I find anything relevant.

**Reply by andreipoe** [reply]
Hello. I built a new Docker image based on the latest commit and I now saw an error in the logs when I tried to sign in: `ValidationError [SequelizeValidationError]: Validation error: Validation isIn on language failed`. It seems the issue was I had set `DEFAULT_LANGUAGE=en_GB`, but only `en_US` is supported. Once I reverted that, I was able to log in. 

I still got an `InvalidAccessKeyId: The AWS Access Key Id you provided does not exist in our records` in the logs, but that didn't prevent me from logging in.

**Comment by Freundschaft** [comment, marked as answer]
Just wanted to check in to let everyone know that I solved the issues, apparently there was an issue with SECRET_KEY and UTILS_SECRET not set correctly in my environment, leading to this error
